src/Agent/NewRelic/Agent/Core/Transformers/CustomErrorDataTransformer.cs

```diff
@@ -65,7 +65,10 @@ public void Transform(ErrorData errorData, float priority, string userid)
             // so it can display the stack trace. 
             _attribDefs.TransactionNameForError.TrySetValue(attribValues, errorData.Path);
 
-            _attribDefs.EndUserId.TrySetValue(attribValues, userid);
+            if (!string.IsNullOrEmpty(userid))
```

Comment:
We might want to change it to something like the following.
```suggestion
            if (!string.IsNullOrWhitespace(userid))
```

Reply:
I prefer IsNullOrWhitespace as well.

Reply:
Done. Plus in a couple of other places as well